tidy3d/components/scene.py

```diff
@@ -246,8 +248,25 @@ def intersecting_media(
             mediums = {medium for medium, _ in structures_merged}
             return mediums
 
-        # if the test object is a volume, test each surface recursively
-        surfaces = test_object.surfaces_with_exclusion(**test_object.dict())
+        # If the test object is a volume, test each surface recursively
+        # and exclude surfaces for 2D simulations where one dimension is ignored.
+        exclude_surfaces_2d: List[str] = []
+        if zero_dim_ind is not None:
+            exclude_surfaces_2d = {0: ["x-", "x+"], 1: ["y-", "y+"], 2: ["z-", "z+"]}.get(
+                zero_dim_ind, []
```

Comment:
I suppose this could just be a list? 

```py
[["x-", "x+"], ["y-", "y+"], ["z-", "z+"]][zero_dim_ind]
```

Also if the zero_dim_ind is not found, we probably just want to error here right? instead of returning an empty list?